src/main/scala/viper/gobra/ast/frontend/Ast.scala

```diff
@@ -416,12 +423,33 @@ case class PExpCompositeVal(exp: PExpression) extends PCompositeVal // exp is ne
 
 case class PLitCompositeVal(lit: PLiteralValue) extends PCompositeVal
 
-case class PFunctionLit(args: Vector[PParameter], result: PResult, body: PBlock) extends PLiteral with PCodeRootWithResult with PScope
+case class PFunctionLit(decl: PClosureNamedDecl) extends PLiteral with PActualExpression
+
+case class PClosureNamedDecl(id: Option[PIdnDef], decl: PClosureDecl) extends PGhostMisc {
+  val result: PResult = decl.result
+}
```

Comment:
this name might be a bit misleading: I believe that all other nodes that have "Named" in the name have a non-optional identifier of type `PIdnDef`. Additionaly, the extra indirection through `decl` is maybe unnecessary. Maybe one could make the class structure a bit simpler and follow the patterns already existent in the codebase (e.g., with `PNamedParameter` and `PUnnamedParameter`) with the following code:

```scala
// replaces PFunctionLit and PClosureDecl
sealed trait PClosureDecl extends PLiteral with PCodeRootWithResult { // "extends PActualExpression" is redundant here
    def args: Vector[PParameter]
    def result: PResult
    def spec: PFunctionSpec
    def body: Option[(PBodyParameterInfo, PBlock)])
}

// replaces PClosureNamedDecl when id != None
case class PNamedClosureDecl(
    id: PIdnDef,
    override val args: Vector[PParameter],
    override val result: PResult
    override val spec: PFunctionSpec
    override val body: Option[(PBodyParameterInfo, PBlock)]) extends PClosureDecl

// replaces PClosureNamedDecl when id == None
case class PUnnamedClosureDecl(
    override val args: Vector[PParameter],
    override val result: PResult
    override val spec: PFunctionSpec
    override val body: Option[(PBodyParameterInfo, PBlock)]) extends PClosureDecl
```
Additionally, this avoids making closure declarations subtypes of `PMisc` which I think makes more sense.

Reply:
One indirection is (almost) necessary: the id of the literal needs to be outside of the scoped node, so that it can be referenced outside (as spec). Without this indirection, we would need to add ad-hoc rules in the Name Resolution, which I think would be worse. 

The second indirection was there so that the id of the literal would be considered ghost, so that it cannot be used as a normal closure in actual expressions. But it is possible to do without too much additional effort, so I have opted for removing the second indirection: there is now only `PFunctionLit` and `PClosureDecl`.

Hopefully the two remaining names are clear enough. PClosureDecl is still a `Misc`, but at least there is only one